W-13796916 import project duke (#91)
mulesoft/docs-code-builder: modules/ROOT/pages/upload-a-project.adoc

= Import a Project into an IDE Workspace
//variable used in multiple places on this page:
:root-dir: This directory is the top-level folder that Anypoint Code Builder generates when it creates a project. Do not navigate to the project’s parent or ancestor directory or to one of its child or descendant directories.
//variable used in multiple places on this page:

> :open-root: If you do not open the root directory, the project does not load correctly because Visual Studio Code does not recognize the directory as a workspace, so the UI canvas for the Mule application does not load properly.

Reviewer: ```suggestion
:open-root: If you do not open the root directory, the UI canvas for your project does not load correctly because Visual Studio Code does not recognize the directory as a workspace.
```

Visual Studio Code treats a project directory as a workspace when you open it to the project's root directory or if your project folder is located in a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^]. 

== Import a Project into a Desktop IDE Workspace

The desktop IDE enables you to upload a project directory or package JAR file into a workspace. 

* <<desktop-project-folder>>
* <<desktop-project-package>>

[[desktop-project-folder]]
=== Import a Project Folder into a Desktop IDE Workspace

. Open the Command Palette by pressing Ctrl+Shift+p (Windows) or Cmd+Shift+p (Mac), and provide the following command:
+
[source,command]
----
File: Open Folder...
----
. Navigate to the _root_ directory for your project. 
+
// see text for this variable defined under title
{root-dir}
. Open the root directory for your project.
+
// see text for this variable defined under title
{open-root}

[[desktop-project-package]]
=== Import a Packaged Project into a Desktop IDE Workspace

> Import a JAR file and automatically unpack the contents to a project folder.

Author: ```suggestion
Import a JAR file to automatically unpack its contents to a project folder.
```

> Import and unpack a project JAR file into a workspace.

Reviewer 2: There is no need to unpack anything. ACB import functionality will unpack it

> Import and unpack a project JAR file into a workspace.

Author: ```suggestion
Import the JAR file.
+
This process automatically unpacks the contents to a project folder.
```

> When you import a JAR file, Anypoint Code Builder automatically unpacks the contents to a project folder.

Author: ```suggestion
Import a JAR file and automatically unpack the contents to a project folder.
```

> . Open the Command Palette by pressing Ctrl+Shift+p (Windows) or Cmd+Shift+p (Mac), and provide the following command:

Reviewer: I think the steps for opening the cmd palette were a bit different maybe Ctrl/Cmd+Shift+p. Just something to think about.

Author: @glenn-rodgers-sf The style guidelines require that formatting and way of describing this. Not my decision to make here. I'm not sure it's always super clear, but not my call ¯\_(ツ)_/¯

+
[source,command]
----
MuleSoft: Import a Project
----
. Select *Mule Project Type*:
+
image::import-mule-project.png[]

> . Navigate to the JAR file and click *Select a jar file to import*.

Reviewer: ```suggestion
. Navigate to the JAR file, and click *Select a jar file to import*.
```

I believe imperative fanboy imperative should be separated by a comma.

> . Navigate to a parent directory into which you want to store the decompressed project folder and click *Select a folder to work*.

Reviewer: ```suggestion
. Navigate to a parent directory into which you want to store the decompressed project folder, and click *Select a folder to work*.
```

= Import a Project into an IDE Workspace
//variable used in multiple places on this page:
:root-dir: This directory is the top-level folder that Anypoint Code Builder generates when it creates a project. Do not navigate to the project’s parent or ancestor directory or to one of its child or descendant directories.
//variable used in multiple places on this page:
:open-root: If you do not open the root directory, the UI canvas for your project does not load correctly because Visual Studio Code does not recognize the directory as a workspace.

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//LOGO (web, desktop, or both)
include::partial$acb-ide-logos.adoc[tags="both-ides"]

Import a project directory or a packaged project into a workspace in your IDE. A project directory contains an API specification or a Mule application for an implementation or integration. 

Visual Studio Code treats a project directory as a workspace when you open it to the project's root directory or if your project folder is located in a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^]. 

> To _export_ a project JAR file package, see xref:package-mule-application.adoc[].

Author: ```suggestion
To _export_ a JAR file that packages a project with your application, see xref:package-mule-application.adoc[].
```

The desktop IDE enables you to upload a project directory or package JAR file into a workspace. 

* <<desktop-project-folder>>
* <<desktop-project-package>>

[[desktop-project-folder]]
=== Import a Project Folder into a Desktop IDE Workspace

. Open the Command Palette by pressing Ctrl+Shift+p (Windows) or Cmd+Shift+p (Mac), and provide the following command:
+
[source,command]
----
File: Open Folder...
----
. Navigate to the _root_ directory for your project. 
+
// see text for this variable defined under title
{root-dir}
. Open the root directory for your project.
+
// see text for this variable defined under title
{open-root}

[[desktop-project-package]]
=== Import a Packaged Project into a Desktop IDE Workspace

> Import the JAR file.

Author: ```suggestion
Import the JAR file.
```

> Import the JAR file.
> +
> This process automatically unpacks the contents to a project folder.

Author: ```suggestion
When you import a JAR file, Anypoint Code Builder automatically unpacks the contents to a project folder.
```

> Import a JAR file to automatically unpack its contents to a project folder.

Author: ```suggestion
Import a JAR file and load the unpacked project to a workspace in the IDE.
```

----
File: Open Folder...
----

> 

Author: @pablocabrera85 Can you look at the section below and approve if you are ok with this?

. Select your Anypoint Code Builder home directory:
+
image::select-home-directory.png[]
. After Anypoint Code Builder loads your home directory, drag your project directory to your Explorer view:
+
image::drag-folder-explorer.png[]
+
If prompted, trust the authors of your file or directory:
+
image::trust-home-directory.png[]
+
Anypoint Code Builder displays an upload progress bar that you can click to expand.

. To load the project into a workspace, open the _root_ directory for the imported project folder.
+
// see text for this variable defined under title
{open-root}

.. Open the Command Palette by pressing Ctrl+Shift+p (Windows) or Cmd+Shift+p (Mac), and provide the following command:
+
[source,command]
----
File: Open Folder...
----

> .. Provide the path to the root directory of the project and click *OK* to load the project root directory into the workspace.

Reviewer: ```suggestion
.. Provide the path to the root directory of the project, and click *OK* to load the project root directory into the workspace.
```

//WEB: PACKAGED PROJECT TO WORKSPACE
[[web-project-package]]
=== Import a Packaged Project into a Web IDE Workspace

> Import and unpack a project JAR file for a project:

Reviewer 2: Same as before, unpack is done by ACB

> Import and unpack a project JAR file for a project:

Author: When you import a JAR file, Anypoint Code Builder automatically unpacks the contents to a project folder.

> Import and unpack a project JAR file for a project:

Author: ```suggestion
When you import a JAR file, Anypoint Code Builder automatically unpacks the contents to a project folder.
```

> Import a JAR file and automatically unpack the contents to a project folder.

Author: ```suggestion
Import a JAR file to automatically unpack its contents to a project folder.
```

> When you import a JAR file, Anypoint Code Builder automatically unpacks the contents to a project folder.

Author: ```suggestion
Import a JAR file and automatically unpack the contents to a project folder.
```

. Select your Anypoint Code Builder home directory:
+
image::select-home-directory.png[]
. After Anypoint Code Builder loads your home directory, drag your project directory to your Explorer view:
+
image::drag-folder-explorer.png[]
+
If prompted, trust the authors of your file or directory:
+
image::trust-home-directory.png[]
+
Anypoint Code Builder displays an upload progress bar that you can click to expand.

. To load the project into a workspace, open the _root_ directory for the imported project folder.
+
// see text for this variable defined under title
{open-root}

.. Open the Command Palette by pressing Ctrl+Shift+p (Windows) or Cmd+Shift+p (Mac), and provide the following command:
+
[source,command]
----
File: Open Folder...
----
.. Provide the path to the root directory of the project, and click *OK* to load the project root directory into the workspace.

//WEB: PACKAGED PROJECT TO WORKSPACE
[[web-project-package]]
=== Import a Packaged Project into a Web IDE Workspace

> Import a JAR file to automatically unpack its contents to a project folder.

Author: ```suggestion
Import a JAR file and load the unpacked project to a workspace in the IDE.
```
